keras_contrib/applications/resnet.py

```diff
@@ -173,99 +227,157 @@ def _block_name_base(stage, block):
 
 
 def basic_block(filters, stage, block, transition_strides=(1, 1),
-                dilation_rate=(1, 1), is_first_block_of_first_layer=False, dropout=None,
-                residual_unit=_bn_relu_conv):
+                dilation_rate=(1, 1), is_first_block_of_first_layer=False,
+                dropout=None, residual_unit=_bn_relu_conv,
+                time_distributed=False, verbose=False):
     """Basic 3 X 3 convolution blocks for use on resnets with layers <= 34.
     Follows improved proposed scheme in http://arxiv.org/pdf/1603.05027v2.pdf
     """
     def f(input_features):
+        if verbose:
+            print("    basic block")
         conv_name_base, bn_name_base = _block_name_base(stage, block)
         if is_first_block_of_first_layer:
             # don't repeat bn->relu since we just did bn->relu->maxpool
-            x = Conv2D(filters=filters, kernel_size=(3, 3),
-                       strides=transition_strides,
-                       dilation_rate=dilation_rate,
-                       padding="same",
-                       kernel_initializer="he_normal",
-                       kernel_regularizer=l2(1e-4),
-                       name=conv_name_base + '2a')(input_features)
+            if time_distributed:
+                x = TimeDistributed(Conv2D(filters=filters, kernel_size=(3, 3),
```

Comment:
same as above, change to a single convolution and add the TimeDistributed component below. Will the two calls to TimeDistributed interfere with each other?

Reply:
Great suggestion, will implement and squash commits.